synapse/handlers/receipts.py

```diff
@@ -203,20 +203,27 @@ def filter_out_private_receipts(
             for event_id, orig_event_content in room.get("content", {}).items():
                 event_content = orig_event_content
                 # If there are private read receipts, additional logic is necessary.
-                if ReceiptTypes.READ_PRIVATE in event_content:
+                if (
+                    ReceiptTypes.READ_PRIVATE in event_content
+                    or "org.matrix.msc2285.read.private" in event_content
```

Comment:
Please let's make this a separate constant, something like:

```suggestion
                    or ReceiptTypes.READ_PRIVATE_UNSTABLE in event_content
```